### PR TITLE
Index locationid column in devices table (CU-86dtm6em1)

### PR DESCRIPTION
**Pull request by elanawood**
Added a migration script that created an index on the locationid column in the devices table.

### CODE REVIEW

File: server/db/045-addIndexToLocationId.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    -- Only execute this script if its migration ID is next after the last successful migration ID
+    IF migrationId - lastSuccessfulMigrationId = 1 THEN
+    
+        CREATE INDEX IF NOT EXISTS devices_locationid_idx ON devices (locationid);
+
+        -- Update the migration ID of the last file to be successfully run to the migration ID of this file
+        INSERT INTO migrations (id)
+        VALUES (migrationId);
+    END IF;
+END $migration$;
```
Comment (micahdbak) [inline server/db/045-addIndexToLocationId.sql:21]: Nice!

**Review by micahdbak — COMMENTED** [review]
I'll consider this PR approved when the shell scripts under server/db are updated to take in `locationid` in addition to the data already input with CSV files. Make sure to udpate the `UPDATE` SQL query to set the locationid column too.

Also, please deploy these changes to the dev buttons server and say that you did so in the description - i want to make sure the index is still valid even when NULL.

File: server/db/045-addIndexToLocationId.sql (hunk 1 above)
Comment (elanawood) [reply server/db/045-addIndexToLocationId.sql:21]: Committed shell changes!

Comment (elanawood) [reply server/db/045-addIndexToLocationId.sql:21]: deployed!

File: README.md
Comment (Darrickg) [inline README.md:1]: this shouldn't be removed

File: server/package-lock.json
Comment (Darrickg) [inline server/package-lock.json:1]: this also shouldnt be changed

**Review by Darrickg — CHANGES_REQUESTED** [review]
it seems like you pulled from main then reverted the changes in main. i commented on what shouldnt be removed for you to check

**Review by Darrickg — APPROVED** [review]
looks like all the main changes were reverted, feel free to merge!

File: server/db/add_buttons_to_client.sh [hunk 2]
```diff
@@ -47,8 +47,9 @@ else
             echo "  Display Name: $unit"
```
Comment (micahdbak) [inline server/db/add_buttons_to_client.sh:47]: At line 44, you need to change

```
    while IFS=",", read -r unit phone_number button_serial_number; do
```

To

```
    while IFS=",", read -r unit phone_number button_serial_number locationid; do
```

For the locationid variable to be populated

File: server/db/add_client.sh [hunk 3]
```diff
@@ -10,9 +10,9 @@ if [[ $EUID > 0 ]]; then
     exit 1
 elif [[ ! -n "$6" ]]; then
     echo ""
-    echo "Usage: $0 path_to_.env_file new_client_name responder_phone_numbers fallback_phone_numbers from_phone_number path_to_buttons_csv"
+    echo "Usage: $0 path_to_.env_file new_client_name responder_phone_numbers fallback_phone_numbers from_phone_number path_to_buttons_csv locationid"
```
Comment (micahdbak) [inline server/db/add_client.sh:13]: Locationid shouldn't be here; leave this line unchanged

File: server/db/add_client.sh [hunk 4]
```diff
@@ -10,9 +10,9 @@ if [[ $EUID > 0 ]]; then
...
     echo "" 
-    echo "Example: $0 ./../.env NewClient '{+16041234567,+16049876541}' '{+17781234567,+17789876543}' +18881112222 ./add_buttons.csv.example"
+    echo "Example: $0 ./../.env NewClient '{+16041234567,+16049876541}' '{+17781234567,+17789876543}' +18881112222 ./add_buttons.csv.example Library_1"
```
Comment (micahdbak) [inline server/db/add_client.sh:15]: Ditto to the above

File: server/db/add_client.sh [hunk 5]
```diff
@@ -39,7 +39,8 @@ else
     echo "  Responder Phones: $3"
     echo "  Fallback Phones: $4"
     echo "  From Phone: $5"
-    sudo PGPASSWORD=$PG_PASSWORD psql -U $PG_USER -h $PG_HOST -p $PG_PORT -d $PG_USER --set=sslmode=require -c "INSERT INTO clients (display_name, responder_phone_numbers, fallback_phone_numbers, from_phone_number, is_displayed, is_sending_alerts, is_sending_vitals) VALUES ('$2', '$3', '$4', '$5', 't', 'f', 'f');"
+    echo "  Locationid: $6"
+    sudo PGPASSWORD=$PG_PASSWORD psql -U $PG_USER -h $PG_HOST -p $PG_PORT -d $PG_USER --set=sslmode=require -c "INSERT INTO clients (display_name, responder_phone_numbers, fallback_phone_numbers, from_phone_number, locationid, is_displayed, is_sending_alerts, is_sending_vitals) VALUES ('$2', '$3', '$4', '$5', '$6', 't', 'f', 'f');"
```
Comment (micahdbak) [inline server/db/add_client.sh:43]: There isn't a locationid column on the clients table, so these two lines (42, 43) should be unchanged

File: server/db/add_client.sh [hunk 6]
```diff
@@ -49,8 +50,9 @@ else
             echo "  Display Name: $unit"
             echo "  Phone Number: $phone_number"
             echo "  Serial Number: $button_serial_number"
+            echo "  Locationid: $locationid"
 
-            sudo PGPASSWORD=$PG_PASSWORD psql -U $PG_USER -h $PG_HOST -p $PG_PORT -d $PG_USER --set=sslmode=require -c "INSERT INTO devices (serial_number, display_name, phone_number, client_id, is_displayed, is_sending_alerts, is_sending_vitals, device_type) VALUES (LOWER('$button_serial_number'), '$unit', '$phone_number', '$client_id', 't', 'f', 'f', 'DEVICE_BUTTON');"
+            sudo PGPASSWORD=$PG_PASSWORD psql -U $PG_USER -h $PG_HOST -p $PG_PORT -d $PG_USER --set=sslmode=require -c "INSERT INTO devices (serial_number, display_name, phone_number, client_id, locationid, is_displayed, is_sending_alerts, is_sending_vitals, device_type) VALUES (LOWER('$button_serial_number'), '$unit', '$phone_number', '$client_id', '$locationid', 't', 'f', 'f', 'DEVICE_BUTTON');"
```
Comment (micahdbak) [inline server/db/add_client.sh:55]: Here is good, thanks for adding

File: server/db/add_client.sh [hunk 7]
```diff
@@ -49,8 +50,9 @@ else
             echo "  Display Name: $unit"
```
Comment (micahdbak) [inline server/db/add_client.sh:50]: Same as in the add_buttons_to_client.sh file, please change line 48 from

```
    while IFS=",", read -r unit phone_number button_serial_number; do
```

To

```
    while IFS=",", read -r unit phone_number button_serial_number locationid; do
```

For the locationid variable to be populated

**Review by micahdbak — COMMENTED** [review]
Some changes are needed for the shell scripts to work properly!

File: server/db/add_buttons_to_client.sh (hunk 2 above)
Comment (elanawood) [reply server/db/add_buttons_to_client.sh:47]: Done!

File: server/db/add_client.sh (hunk 7 above)
Comment (elanawood) [reply server/db/add_client.sh:49]: done!

File: server/db/add_client.sh (hunk 6 above)
Comment (elanawood) [reply server/db/add_client.sh:54]: slay!

File: server/db/add_client.sh (hunk 4 above)
Comment (elanawood) [reply server/db/add_client.sh]: done!

File: server/db/add_client.sh (hunk 3 above)
Comment (elanawood) [reply server/db/add_client.sh]: fixed this!

**Review by micahdbak — APPROVED** [review]
Looks good - merge it!!!

**Pull request merged**